database.go

```diff
@@ -1075,6 +1134,34 @@ func (db *database) addSeriesToIndex(measurementName string, s *Series) bool {
 	return idx.addSeries(s)
 }
 
+// dropSeries removes the series from the in memory references
+func (db *database) dropSeries(seriesIDs ...uint32) error {
+	for _, id := range seriesIDs {
+		// if the series is already gone, return
+		if db.series[id] == nil {
+			continue
+		}
+
+		delete(db.series, id)
+
+		// Remove series information from measurements
+		for _, m := range db.measurements {
+			if !m.dropSeries(id) {
+				return fmt.Errorf("failed to remove series id %d from measurment %q", id, m.Name)
```

Comment:
the `dropSeries` method on measurement doesn't return an error, but a bool. Just ignore if its already been dropped. That's fine, it's an idempotent operation.